Feature/cleanup build

tomlogic opened the pull request:
@spth, take a look at these changes.  I'm removing files generated by `autoconf` and adding them to the `.gitignore` file so they don't show up in the repository.  My build system modified many of the files, and it isn't necessary to include them in the repository.

You can test building on your system(s) by checking out this branch, running `autoreconf -i`, then `./configure` and `make`.

tomlogic committed:
build: remove auto-generated files

tomlogic committed:
git: ignore autoconf files

tomlogic committed:
coldboot: quiet compiler warning in makecold.cpp

tomlogic committed:
doc: update README with `autoreconf` instructions

tomlogic commented on coldboot/makecold.cpp:
```diff
@@ -180,7 +180,7 @@ int main(int argc, char* argv[])
-   printf("output file successfully created (%d bytes)\n",
-   	x*3+sizeof(start_sequence)-1);
+   printf("output file successfully created (%u bytes)\n",
+   	(int)(x * 3 + sizeof(start_sequence) - 1));
```
I addressed a compiler warning while I was at it.

spth commented:
I do not have strong opinion of this.  I understand there are two approaches to files generated by autotools:

1) Leave them out of the repo - keeps the repo clean
2) Keep them in the repo - makes it easy to build, without requiring a particular autotools version

I went for 2) so far, since that is what the OpenRabbit apparently used before I took over.

If you prefer 1) and want to remove the ./configure file I suggest:
1) Add a ./bootstrap script that invokes autoreconf -i, so users don't need to really know we use autotools.
2) Just unconditionally depend on SDCC and build coldload.bin. This makes it a bit harder for users that just want to use files from Dynamic C, but I guess those are few, and it simplifies the build infrastructure.

tomlogic commented:
My preference is to go with these changes:

1. The current setup failed for me until I installed updated tools.  `./configure` was complaining that `aclocal-1.16` wasn't available.  So I'm hesitant to believe that keeping the files in the repo makes it any easier to build.
2. I might have been able to use an older version of `aclocal` had I known about `autoreconf`.  As it was, the existing build files generated with aclocal 1.16 made it appear to be a requirement for the project.
3. Multiple developers working in the repository will mean constant changes to those files, which means more work when making commits.  I had git informing me of changes to:
    * Makefile.in
    * aclocal.m4
    * coldboot/Makefile.in
    * config.h.in
    * configure
    * src/Makefile.in
4. I updated the `README` with instructions on running `autoreconf -i` to generate the `configure` script.  I'm not sure that a `bootstrap` script with that one line in it will be easier to follow.  We could rename `README` to `BUILD` for clarity.
5. I'd like to keep the SDCC requirement optional, and have a `coldload.bin` available with `pilot.bin` for people who just want to build the loader.  Does `autoreconf` fail if SDCC isn't present?  Is it possible to update `configure.ac` to mark it as an optional component?

tomlogic commented:
Do we want to add a copy of `coldload.bin` to the repository for people who aren't going to build it themselves?

spth commented:
The current README contains some information on how to build, but most of the content is other stuff, so I wouldn't want to rename it.
Currently, the build of coldload.bin is optional. In trunk, and I think also in this branch. SDCC is required only if coldload.bin is built.

spth commented:
> Do we want to add a copy of `coldload.bin` to the repository for people who aren't going to build it themselves?

I don't think so. I think the only reason not to build it is not having SDCC installed. And those who don't have SDCC installed will use OpenRabbit with Dynamic C, so they could get their coldload.bin from there.

tomlogic commented:
How do you feel about merging this as-is?  You said you don't have a strong opinion, so is my opinion strong enough to sway you into merging?  Would it help if I rebased on `HEAD` and split the build instructions out into a `BUILD` or `BUILDING` file (and potentially reference that from the `README`)?
